### ISSUE TITLE
While cached image is loading, CachedNetworkImage has no size

### ISSUE THREAD

**Issue by neilweber** [renefloor/flutter_cached_network_image#328]
I switched from FadeInImage.assetNetwork() to CachedNetworkImage in a ListView and everything appears to be working well.  Except I noticed that after the image has been cached, the next time I launched my app to redisplay the list, all of the list tiles started out very short even though I was specifying a height (of 200.0).  Once the image is loaded, the list tile is sized as specified.

As a work around, I enclosed the CachedNetworkImage in a Container and now the list is working as desired again:

```
              child: Container(
                width: double.infinity,
                height: 200.0,
                child: CachedNetworkImage(
                  placeholder: (context, url) => Image.asset('assets/placeholder.jpg'),
                  imageUrl: movie.getBackDropUrl(),
                  fit: BoxFit.cover,
                  width: double.infinity,
                  height: 200.0,
                  fadeInDuration: Duration(milliseconds: 50),
                ),
              ),
```

**Comment by renefloor** [comment]
I just had a look at the code and it seems to be that the widget doesn't use the height when you have a placeholder. If you have no placeholder it will size to the defined width and height, but otherwise it will take the height of the placeholder itself. Could it be that your placeholder.jpg is very small?

**Comment by neilweber** [comment]
My placeholder is actually huge: 512 pixels tall.  Here is my project: https://github.com/neilweber/flutter_cinematic and the file where I noticed this issue is https://github.com/neilweber/flutter_cinematic/blob/master/lib/widgets/media_list/media_list_item.dart.  If you want, I can try to recreate the problem in stand-alone app, though it may not be done for a few days.  Let me know.

**Comment by renefloor** [comment]
@neilweber it would be nice if you can do that. It's easier fixing for me when I have a concrete example that I can run.

**Comment by SpajicM** [comment]
Same issue here, causing pretty ugly layout jumps.

**Comment by hlongngo** [comment]
I got the same issue.

This is inside ListView itemBuilder
`return CachedNetworkImage(
        imageUrl: url,
        placeholder: (BuildContext context, String url) {
          return Container(
            width: MediaQuery.of(context).size.width,
            height: MediaQuery.of(context).size.height,
            child: Center(
              child: CircularProgressIndicator(),
            ),
          );
        },
        width: MediaQuery.of(context).size.width,
        fit: BoxFit.fitWidth,
      );`

CachedNetworkImage has no size (or very short) when image is loading

**Comment by montyr75** [comment]
Yeah, my cached images load centered in the frame, then jump left. It's very annoying...

**Comment by renefloor** [comment]
> Yeah, my cached images load centered in the frame, then jump left. It's very annoying...

@montyr75 do you set a height and width on the image widget? Otherwise before loading it can't know the size and after loading it makes the widget the size of the image. If you want to keep it centered you can put it in a Center() widget.
